comp/otelcol/otlp/components/exporter/datadogexporter/factory.go

```diff
@@ -0,0 +1,199 @@
+		QueueSettings: exporterhelper.NewDefaultQueueSettings(),
+
+		API: APIConfig{
+			Site: "datadoghq.com",
+		},
+
+		Metrics: serializerexporter.MetricsConfig{
+			DeltaTTL: 3600,
+			ExporterConfig: serializerexporter.MetricsExporterConfig{
+				ResourceAttributesAsTags:           false,
+				InstrumentationScopeMetadataAsTags: false,
+			},
+			HistConfig: serializerexporter.HistogramConfig{
+				Mode:             "distributions",
+				SendAggregations: false,
+			},
+			SumConfig: serializerexporter.SumConfig{
+				CumulativeMonotonicMode:        serializerexporter.CumulativeMonotonicSumModeToDelta,
+				InitialCumulativeMonotonicMode: serializerexporter.InitialValueModeAuto,
+			},
+			SummaryConfig: serializerexporter.SummaryConfig{
+				Mode: serializerexporter.SummaryModeGauges,
+			},
+		},
+
+		Traces: TracesConfig{
+			TCPAddrConfig: confignet.TCPAddrConfig{
+				Endpoint: "https://trace.agent.datadoghq.com",
+			},
+			IgnoreResources: []string{},
+		},
+
+		Logs: LogsConfig{
+			TCPAddrConfig: confignet.TCPAddrConfig{
+				Endpoint: "https://http-intake.logs.datadoghq.com",
+			},
+		},
+
+		HostMetadata: HostMetadataConfig{
+			Enabled:        true,
+			HostnameSource: HostnameSourceConfigOrSystem,
+		},
+	}
+}
+
+// checkAndCastConfig checks the configuration type and its warnings, and casts it to
+// the Datadog Config struct.
+func checkAndCastConfig(c component.Config, logger *zap.Logger) *Config {
+	cfg, ok := c.(*Config)
+	if !ok {
+		panic("programming error: config structure is not of type *datadogexporter.Config")
+	}
+	cfg.logWarnings(logger)
+	return cfg
+}
+
+// createTracesExporter creates a trace exporter based on this config.
+func (f *factory) createTracesExporter(
+	_ context.Context,
+	_ exporter.CreateSettings,
+	_ component.Config,
+) (exporter.Traces, error) {
+	// TODO implement
+	return nil, nil
+}
+
+// createTracesExporter creates a trace exporter based on this config.
+func (f *factory) createMetricsExporter(
+	ctx context.Context,
+	set exporter.CreateSettings,
+	c component.Config,
+) (exporter.Metrics, error) {
+	cfg := checkAndCastConfig(c, set.Logger)
+	sf := serializerexporter.NewFactory(f.s, &tagEnricher{}, f.h.Get)
+	ex := &serializerexporter.ExporterConfig{
+		Metrics: cfg.Metrics,
+		TimeoutSettings: exporterhelper.TimeoutSettings{
+			Timeout: cfg.Timeout,
+		},
+		QueueSettings: cfg.QueueSettings,
+	}
+	return sf.CreateMetricsExporter(ctx, set, ex)
+}
+
+// createLogsExporter creates a logs exporter based on the config.
+func (f *factory) createLogsExporter(
+	ctx context.Context,
+	set exporter.CreateSettings,
+	_ component.Config,
+) (exporter.Logs, error) {
+	var logch chan *message.Message
+	if provider := f.logsAgent.GetPipelineProvider(); provider != nil {
+		logch = provider.NextPipelineChan()
+	}
+	lf := logsagentexporter.NewFactory(logch)
+	lc := &logsagentexporter.Config{
+		OtelSource:    "otel_agent",
+		LogSourceName: "otelcol",
```

Comment:
This should be set to `logsagentexporter.logSourceName` ("OTLP log ingestion") to be consistent